lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -170,16 +171,20 @@ def _on_leader_elected(self, event: LeaderElectedEvent):
 
         if not self.peers_data.get(Scope.APP, "admin_user_initialized"):
             self.unit.status = MaintenanceStatus(AdminUserInitProgress)
-            # User config is currently in a default state, which contains multiple insecure default
-            # users. Purge the user list before initialising the users the charm requires.
-            self._purge_users()
-            self._put_admin_user()
-            self.peers_data.put(Scope.APP, "admin_user_initialized", True)
-            self.status.clear(AdminUserInitProgress)
+
+        # User config is currently in a default state, which contains multiple insecure default
+        # users. Purge the user list before initialising the users the charm requires.
+        self._purge_users()
+
+        # this is in case we're coming from 0 to N units, we don't want to use the rest api
+        self._put_admin_user(
+            keep_current_if_exists=self.peers_data.get(Scope.APP, "admin_user_initialized", False)
+        )
```

Comment:
you don't need arg, you can check `admin_user_initialized` inside `_put_admin_user`
```suggestion
        self._put_admin_user()
```

Reply:
Good idea, thank you